app/javascript/packages/document-capture/components/suspense-error-boundary.tsx

```diff
@@ -0,0 +1,63 @@
+import { Component, Suspense } from 'react';
+import type { ReactNode } from 'react';
+import { RetrySubmissionError } from './submission-complete';
+
+interface SuspenseErrorBoundaryProps {
+  /**
+   * Fallback to show while suspense pending.
+   */
+  fallback: NonNullable<ReactNode> | null;
+  /**
+   * Error callback.
+   */
+  onError: (error: Error) => void;
+  /**
+   * Error instance caught to allow for acknowledgment of rerender, in order to prevent infinite rerendering.
+   */
+  handledError?: RetrySubmissionError;
```

Comment:
Could we use the more generic `Error` superclass, since it doesn't specifically need to be this subclass?

```suggestion
  handledError?: Error;
```